From 2c830f16f8be3e2a980c948c536c02062171a830 Mon Sep 17 00:00:00 2001
From: Bjorn Andersson <bjorn.andersson@linaro.org>
Date: Tue, 7 Dec 2021 17:59:43 -0800
Subject: [PATCH 100/230] hack: disable has_address_auth_metacap and
 has_generic_auth

Keeping these enabled prevents the sc8280xp from booting.

Signed-off-by: Bjorn Andersson <bjorn.andersson@linaro.org>
---
 arch/arm64/kernel/cpufeature.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm64/kernel/cpufeature.c b/arch/arm64/kernel/cpufeature.c
index a77315b338e..c91b04ecd77 100644
--- a/arch/arm64/kernel/cpufeature.c
+++ b/arch/arm64/kernel/cpufeature.c
@@ -2012,6 +2012,7 @@ static bool has_address_auth_metacap(const struct arm64_cpu_capabilities *entry,
 	bool apa = has_address_auth_cpucap(cpu_hwcaps_ptrs[ARM64_HAS_ADDRESS_AUTH_ARCH_QARMA5], scope);
 	bool apa3 = has_address_auth_cpucap(cpu_hwcaps_ptrs[ARM64_HAS_ADDRESS_AUTH_ARCH_QARMA3], scope);
 
+	return false;
 	return apa || apa3 || api;
 }
 
@@ -2022,6 +2023,7 @@ static bool has_generic_auth(const struct arm64_cpu_capabilities *entry,
 	bool gpa = __system_matches_cap(ARM64_HAS_GENERIC_AUTH_ARCH_QARMA5);
 	bool gpa3 = __system_matches_cap(ARM64_HAS_GENERIC_AUTH_ARCH_QARMA3);
 
+	return false;
 	return gpa || gpa3 || gpi;
 }
 #endif /* CONFIG_ARM64_PTR_AUTH */
-- 
2.39.0

